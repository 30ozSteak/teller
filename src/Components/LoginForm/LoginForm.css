.form-wrapper {
  text-align: center;
}

.form-wrapper input {
  display: block;
  background-color: transparent;
  border: none;
  color: white;
  font-size: 1rem;
  text-align: center;
}

.form-wrapper input:first-child {
  border-bottom: 1px solid var(--TellerPink);
}

.login-form input {
  padding: 0.5rem 1rem;
}

.form-wrapper .login-button {
  border: none;
  display: inline;
}

.play-button {
  width: 15px;
  height: 15px;
}
