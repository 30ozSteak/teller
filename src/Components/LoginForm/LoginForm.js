import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }

  handleSubmit = async (e) => {
    e.preventDefault()
    this.props.logInUser(this.state.email)
  }

  handleChange = async (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { email, password } = this.state
    return (
      <div className="form-wrapper">
        <form className="login-form" onSubmit={this.handleSubmit}>
          <input
            className="form-input user-email"
            name="email"
            value={email}
            onChange={this.handleChange}
            placeholder="Enter Email Address"
          />
          <input
            className="form-input user-password"
            type="password"
            name="password"
            value={password}
            onChange={this.handleChange}
            placeholder="Enter Password"
          />
          <button className="login-button">Login</button>
        </form>
      </div>
    )
  }
}

export default LoginForm
