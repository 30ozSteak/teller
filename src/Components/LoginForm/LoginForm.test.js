import React from 'react'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import LoginForm from './LoginForm'
import { shallow } from 'enzyme'

describe('LoginForm Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<LoginForm />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
