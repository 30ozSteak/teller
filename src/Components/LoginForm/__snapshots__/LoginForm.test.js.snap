// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component should match the snapshot 1`] = `undefined`;

exports[`LoginForm matches the snapshot 1`] = `undefined`;
